web/src/types/index.ts

```diff
@@ -1,28 +1,75 @@
 export type App = {
   allowSnapshots: boolean;
+  autoDeploy: boolean;
   chartPath: string;
   credentials: Credentials;
   currentSequence: number;
-  downstream: DownStream;
+  downstream: Downstream;
   hasPreflight: boolean;
+  id: string;
+  iconUri: string;
+  isAirgap: boolean;
   isConfigurable: boolean;
   isGeoaxisSupported: boolean;
   isGitOpsSupported: boolean;
   isIdentityServiceSupported: boolean;
+  isSemverRequired: boolean;
   name: string;
   namespace: string;
   needsRegistry?: boolean;
   slug: string;
+  updateCheckerSpec: string;
+};
+
+export type AppLicense = {
+  assignee: string;
+  channelName: string;
+  entitlements: string[];
+  expiresAt: string;
+  id: string;
+  isAirgapSupported: boolean;
+  isGeoaxisSupported: boolean;
+  isGitOpsSupported: boolean;
+  isIdentityServiceSupported: boolean;
+  isSemverRequired: boolean;
+  isSnapshotsSupported: boolean;
+  isSupportBundleUploadSupported: boolean;
+  lastSyncedAt: string;
+  licenseSequence: number;
+  licenseType: string;
 };
 
 export type Credentials = {
   username: string;
   password: string;
 };
 
-export type DownStream = {
+export type ResourceStates = {
+  kind: string;
+  name: string;
+  namespace: string;
+  // from https://github.com/replicatedhq/kots/blob/84b7e4e0e9275bb200a36be69691c4944eb8cf8f/pkg/appstate/types/types.go#L10-L14
```

Comment:
this is the only time i've looked at the go code to get an enum. there aren't many available but anytime you see a hardcoded string like "ready" being used it's a good idea to check.